pkg/apiclient/application/application.pb.go

```diff
@@ -44,13 +43,13 @@ type ApplicationQuery struct {
 	// forces application reconciliation if set to true
 	Refresh *string `protobuf:"bytes,2,opt,name=refresh" json:"refresh,omitempty"`
 	// the project names to restrict returned list applications
-	Projects []string `protobuf:"bytes,3,rep,name=project" json:"project,omitempty"`
+	Projects []string `protobuf:"bytes,3,rep,name=projects" json:"projects,omitempty"`
```

Comment:
FYI, this change deleted all of our apps since we used `project=XX`. The filter on project was not there and resulted in all our apps being listed and deleted. This was not in the changelog.

Reply:
@victorboissiere would you mind opening an issue requesting that the upgrade guide be changed to include a note about this?

We probably also need to establish a stricter policy about API backwards-compatibility.

Reply:
> We probably also need to establish a stricter policy about API backwards-compatibility.

I think we already promise backwards compatibility for the API. And I think this one just slipped through the cracks.

Reply:
This file is auto-generated. The breaking change [originated here](https://github.com/argoproj/argo-cd/pull/8929/files#diff-df6e82b174b5eaec332889c540b09c2fa66b2eb2bcd1b40545cfd1d2d1800bc2R23). This was probably a misunderstanding about how the `gogoproto.customname` worked. Unfortunately this wasn't caught by our automated tests on that occasion. I agree with @jannfis that a stricter policy isn't the answer to this incident. Our API is exposed at `/api/v1` and that shouldn't break. What we could do to avoid this type of problem in the future is improving our automated tests. Probably having some sort of contract testing would be beneficial.

Reply:
@victorboissiere I am sorry about the incident that this change caused. To avoid this type of issue to happen again in the future, I created the following proposal: https://github.com/argoproj/argo-cd/issues/12589

Reply:
@victorboissiere what was your client? I'm investigating whether this is a problem with the CLI.

Reply:
Thanks for the issue! it was not the CLI, we’ve built an internal tool that cleanup from time to time some project applications. The filter stopped working and we’ve lost ~8000 pods. 
Thanks for the effort to avoid this in the future 🙏

Reply:
Sweet, thanks! PR to fix the issue: https://github.com/argoproj/argo-cd/pull/12594

I had written up some warnings about the CLI, but after investigation, I don't think the CLI is impacted. Only the JSON API.